sycl/doc/SYCLPluginInterface.md

```diff
@@ -0,0 +1,178 @@
+OpenCL implementations. They can be installed either in the standard Khronos
+ICD-compatible way or not, and the OpenCL plugin can hook up with both.
+
+##### Linking to ICD-compatible OpenCL installations
+
+The plugin dynamically loads the OCL ICD loader - libOpenCL.so from system
+paths, and on success it asks the loader to provide all OpenCL implementations
+installed on the system. To do that it finds and calls the "clGetPlatformIDs"
+in the ICD loader and records the returned platform objects for later use. The
+plugin may choose to dynamically link with libOpenCL.so at build time.
+ 
+##### Linking to ICD-incompatible OpenCL installations
+
+For the OpenCL plugin to hook up such installation, it must be described in a
+configuration file similarly to plugin configuration file. *TBD - location and
+format of custom OpenCL installation config file*. The plugin will dynamically
+load the dynamic library pointed to in the configuration file and query it for
+OpenCL API function addresses similarly ICD loader. In more details, it will:
+- search for  the ```clGetExtensionFunctionAddress``` function in the 
+  library and call it with the string *“clIcdGetPlatformIDsKHR”* to find
+  the ```clIcdGetPlatformIDsKHR``` function pointer
+- then, using ```clIcdGetPlatformIDsKHR```, find the set of
+  ```cl_platform_id``` objects that the library provides, and store them
+  internally to enumerate available devices when required
+
+### Device enumeration by plugins
+
+TBD
+
+## PI API Specification
+
+PI allows for two levels of API implementation - basic and extended. The basic
+level comprises core APIs needed to perform SYCL offload. The extended level is
+needed only for OpenCL interoperability. 
+
+### The basic PI APIs
+
+This level requires availability of functions representing core functionality,
+such as device memory management, kernel creation and parameter setting,
+enqueuing kernel for execution, etc. There are functions which fully match
+semantics of the corresponding OpenCL X.X functions, as well as functions which
+don't have direct mapping to OpenCL. See the
+[pi.h](../include/CL/sycl/detail/pi.h) header for the full list and
+descriptions of PI APIs. [TBD: link to pi.h doxygen here]
+
+*__Examples of PI OpenCL-based APIs__*
+
+    piCreateKernel
+    piReleaseKernel
+    piSetKernelArg
+
+*__Example of PI non OpenCL-based APIs__*
+
+Selects the most appropriate device image based on runtime
+information and the image characteristics
+```
+cl_int piSelectDeviceImage(pi_context ctx, pi_device_image **images,
+                             cl_uint num_images,
+                             pi_device_image **selected_image);
+```
+
+PI also defines few types and string tags to describe a device binary image.
+Those are used to communicate to plugins information about the images where it
+is needed, currently only in the ```piSelectDeviceImage``` function. The main
```

Comment:
> PI also defines few types and string tags to describe a device binary image. Those are used ... in the ```piSelectDeviceImage``` function.

I see `pi_device_image` type in `piSelectDeviceImage` declaration, but what is _**string tags**_?

Reply:
@bader,

/// Target identification strings for
/// pi_device_binary_struct.DeviceTargetSpec
///
#define PI_DEVICE_BINARY_TARGET_UNKNOWN "<unknown>"
#define PI_DEVICE_BINARY_TARGET_SPIRV32 "spir"
#define PI_DEVICE_BINARY_TARGET_SPIRV64 "spir64";